File: packages/config/src/servers.ts
export const serverTypeList = [ 'voxcpm_torch_gradio', 'demucs_torch_server',] as const;
export type ServerType = (typeof serverTypeList)[number];

export const SERVICE_MAP: Record<ServerType, string> = {
  voxcpm_torch_gradio: '_ld-voxcpm-py._tcp.local',
  demucs_torch_server: '_ld-demucs-py._tcp.local',
}

File: packages/config-rs/src/servers.rs
/// Server type identifiers for mDNS discovery.
#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
pub enum ServerType {
    VoxcpmTorchGradio,
    DemucsTorchServer,
}

impl ServerType {
    /// All known server types.
    pub const ALL: &'static [ServerType] =
        &[ServerType::VoxcpmTorchGradio, ServerType::DemucsTorchServer];

    /// Corresponding mDNS service name.
    pub fn service_name(self) -> &'static str {
        match self {
            ServerType::VoxcpmTorchGradio => "_ld-voxcpm-py._tcp.local",
            ServerType::DemucsTorchServer => "_ld-demucs-py._tcp.local",
        }
    }
}
